core/templates/dev/head/pages/exploration_editor/statistics_tab/LearnerActionRenderService.js

```diff
@@ -0,0 +1,162 @@
+      return htmlString;
+    };
+
+    var renderLearnerActionsTableForMultipleIncorrectIssue = function(
+        finalBlock) {
+      var index = finalBlock.length - 1;
+      var stateName =
+        finalBlock[index].actionCustomizationArgs.state_name.value;
+
+      var tableHTML =
+        '<table class="learner-actions-table"><tr><th>Answer</th>' +
+        '<th>Feedback</th></tr>';
+      for (var i = 0; i < index; i++) {
+        if (finalBlock[i].actionType !== ACTION_TYPE_ANSWER_SUBMIT) {
+          continue;
+        }
+        var answer = finalBlock[i].actionCustomizationArgs.answer.value;
+        var feedback = finalBlock[i].actionCustomizationArgs.feedback.value;
+        tableHTML +=
+          '<tr><td>' + answer + '</td><td>' + feedback + '</td></tr>';
+      }
+      tableHTML += '</table>';
+      return tableHTML;
+    };
+
+    var renderLearnerActionHTML = function(learnerAction) {
+      var actionType = learnerAction.actionType;
+      var custArgs = learnerAction.actionCustomizationArgs;
+      if (actionType === ACTION_TYPE_EXPLORATION_START) {
+        return renderExplorationStartActionHTML(custArgs.state_name.value);
+      } else if (actionType === ACTION_TYPE_EXPLORATION_QUIT) {
+        return renderExplorationQuitActionHTML(
+          custArgs.state_name.value, custArgs.time_spent_in_state_secs.value);
+      } else if (actionType === ACTION_TYPE_ANSWER_SUBMIT) {
+        interactionId =
+          custArgs.interaction_id.value;
+        if (interactionId === 'Continue') {
+          return renderContinueButtonSubmitActionHTML(
+            custArgs.dest_state_name.value,
+            custArgs.time_spent_in_state_secs.value);
+        }
+        if (custArgs.answer.value) {
+          return renderAnswerSubmitActionHTML(
+            custArgs.answer.value, custArgs.dest_state_name.value,
+            custArgs.time_spent_in_state_secs.value, custArgs.state_name.value
+          );
+        }
+      }
+    };
+
+    return {
+      renderFinalDisplayBlockForMISIssueHTML: function(block) {
+        var index = block.length - 1;
+        var stateName = block[index].actionCustomizationArgs.state_name.value;
+        var htmlString = '';
+        for (
+          var i = 0; block[i].actionType !== ACTION_TYPE_ANSWER_SUBMIT; i++) {
+          htmlString += renderLearnerActionHTML(block[i]);
+        }
+        htmlString +=
+          '<span class="learner-action">Submitted the following answers in ' +
+          'card "' + stateName + '"</span>';
+        htmlString += renderLearnerActionsTableForMultipleIncorrectIssue(block);
+        htmlString += renderLearnerActionHTML(block[index]);
+        return htmlString;
+      },
+      renderDisplayBlockHTML: function(block) {
+        var htmlString = '';
+        for (var i = 0; i < block.length; i++) {
+          htmlString += renderLearnerActionHTML(block[i]);
+        }
+        return htmlString;
+      },
+      getDisplayBlocks: function(learnerActions) {
```

Comment:
I'm a bit confused at why this function is exposed (aside for testing).

Please add to the top-level comment in this file an example of how this service should be used by future playthrough types (so write out simple code showing a how you plan to use this to implement 3.2.2-4).

Reply:
Ive added how this function will be used from the directive. Basically:

1. Directive gets all learner actions split into these display blocks and keeps it there.
2. We display only one block at a time.
3. When we click extend to view the previous block, we convert that particular block into HTML at that instant and then display it.
4. This way, we only convert into HTML what we actually might need to display.